scripts/src/ledger.ts

```diff
@@ -0,0 +1,177 @@
+// Adapted from @solana/wallet-adapter-ledger, with some modifications to support Node.js
+import Transport, {
+  StatusCodes,
+  TransportStatusError,
+} from "@ledgerhq/hw-transport";
+import TransportNodeHid from "@ledgerhq/hw-transport-node-hid";
+import { getDerivationPath } from "@solana/wallet-adapter-ledger";
+import type { Transaction } from "@solana/web3.js";
+import { VersionedTransaction, PublicKey } from "@solana/web3.js";
+
+export function parseDerivationPath(source: string): {
+  derivationAccount?: number;
+  derivationChange?: number;
+} {
+  const params = new URLSearchParams(new URL(source).searchParams);
+  const key = params.get("key");
```

Comment:
I think this is redundant; just do `const key = new URL(source).searchParams.get("key")`